cli/internal/transformer/transformer.go

```diff
@@ -153,9 +154,8 @@ func (t *RecordTransformer) Transform(record arrow.Record) arrow.Record {
 		cols = append(cols, syncGroupIdBldr.NewArray())
 	}
 
-	for i := range sc.Fields() {
-		cols = append(cols, record.Column(i))
-	}
+	cols = cols[:len(sc.Fields())+t.internalColumns] // resize back as we have the capacity
+	copy(cols[t.internalColumns:], record.Columns())
```

Comment:
I might be missing something but we have 3 conditional `append` to `cols` a bit before (also inside `TransformSchema`)
What happens if `t.internalColumns` is `3` for example but we didn't do any `append`.
Won't that mean that we'll have misalignment in the data?

I believe that can happen when someone is syncing to a Postgres DB destination, than using our Postgres source on the synced tables

Reply:
Regardless it would be helpful to understand how the refactor improves the code

Reply:
Ah, I see the issue with the copy here (we may skip append if the data has the col).
I'll fix it with a single append then

Reply:
> Internal columns depends on the amount of added columns.

`internalColumns` is incremented when the option is configured https://github.com/cloudquery/cloudquery/blob/881902078f8293ac5cb374e48c91fddd60b3ab4d/cli/internal/transformer/transformer.go#L38

`cols` is appended based on a slightly different condition https://github.com/cloudquery/cloudquery/blob/881902078f8293ac5cb374e48c91fddd60b3ab4d/cli/internal/transformer/transformer.go#L148